### PR TITLE
Test files and conftest.py

### PR DESCRIPTION
**Pull request by SMoragon**

### CODE REVIEW

File: Tests/my_test_Inference__init_.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            boolvals = [name in clade_term_names for name in term_names]
+            bitstr = _BitString("".join(map(str, map(int, boolvals))))
+            bitstrs.add(bitstr)
+        return bitstrs
+
+    def compare(self, tree1, tree2):
+        term_names1 = [term.name for term in tree1.get_terminals()]
+        term_names2 = [term.name for term in tree2.get_terminals()]
+        # false if terminals are not the same
+        if set(term_names1) != set(term_names2):
```
Comment (nachorequeno) [inline Tests/my_test_Inference__init_.py:39]:
```suggest
        return set(term_names1) == set(term_names2) and self._bitstrs(tree1) == self._bitstrs(tree2)
```

File: Tests/my_test_Inference_FastTree.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        """
+        # random temporary file path generation to save the results of the execution into
+        r = random.randint(1, 999999)
+        treefile_path = Path.joinpath(self.tmp_dir, "FastTree_bestTree." + str(r))
+
+        command += ["-log", treefile_path, infile_path.absolute()]
+        
+        expected_output = pytest.data_dir / expected_output
+        run_mocker = MockStdOut(expected_output, treefile_path)
+        mocked_subprocess_output=run_mocker.get_mocked_output()
```
Comment (nachorequeno) [inline Tests/my_test_Inference_FastTree.py:188]:
```suggest
        mocked_subprocess_output = run_mocker.get_mocked_output()
```

**Review by nachorequeno — CHANGES_REQUESTED** [review]

File: Tests/my_test_Inference_FastTree.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
+from pathlib import Path
+import random
+import subprocess
+import pytest
+from io import StringIO
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_FastTree.py:5]: Missing License header, and the imports should be in alphabetical order.

File: Tests/my_test_Inference_FastTree.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        term_names = [term.name for term in tree.get_terminals()]
+        term_names.sort()
+        for clade in tree.get_nonterminals():
+            clade_term_names = [term.name for term in clade.get_terminals()]
+            boolvals = [name in clade_term_names for name in term_names]
+            bitstr = _BitString("".join(map(str, map(int, boolvals))))
+            bitstrs.add(bitstr)
+        return bitstrs
+
+    def compare(self, tree1, tree2):
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_FastTree.py:40]: Missing type hints and docstring.

File: Tests/my_test_Inference_FastTree.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        return self.stdout
+
+
+class TestInferenceFastTree:
+    tmp_dir: Path = Path("tests/fasttree_tmp_dir/").absolute()
+    manifest_dir: Path
+    # Couple of functions used to compare Phylo Trees:
+
+    # store and return all _BitStrings
+    def _bitstrs(self, tree):
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_FastTree.py:29]: Missing type hints and docstring.

File: Tests/my_test_Inference_FastTree.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+        mocked_subprocess_output=run_mocker.get_mocked_output()
+        #print(mocked_subprocess_output)
+     
+        """
+        output = subprocess.run(
+            command, stderr=subprocess.DEVNULL, universal_newlines=True, check=True, stdout=subprocess.PIPE
+        ).stdout
+        print(r)
+        assert 0
+        """
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_FastTree.py:197]: Why is this here?

File: Tests/my_test_Inference_RAxML.py [hunk 7]
```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+import subprocess
+import pytest
+import random
+import os
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_RAxML.py:5]: Same as before.

File: Tests/my_test_Inference_RAxML.py [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+            boolvals = [name in clade_term_names for name in term_names]
+            bitstr = _BitString("".join(map(str, map(int, boolvals))))
+            bitstrs.add(bitstr)
+        return bitstrs
+
+    def compare(self, tree1, tree2):
+        term_names1 = [term.name for term in tree1.get_terminals()]
+        term_names2 = [term.name for term in tree2.get_terminals()]
+        # false if terminals or BitStrings are not the same
+        return set(term_names1) == set(term_names2) and self._bitstrs(tree1) == self._bitstrs(tree2)
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_RAxML.py:53]: Do not replicate code! If you need this in all your tests, better add this as utils or pytest element.

File: Tests/my_test_Inference_RAxML.py [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+        str_tree=""
+        with open(treefile_path, "r") as treefile:
+            content = treefile.readlines()
+            for line in content:
+                str_tree += line
+        phylogeny = Phylo.read(treefile_path, "newick")
+        res_tree, res_score = Rax.get_results(command, output)
+        assert score == res_score
+        assert str_tree == res_tree.format("newick")
+       """
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference_RAxML.py:264]: Why is this here?

File: Tests/my_test_Inference__init_.py [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+from pathlib import Path
+import random
+from typing import ContextManager
+import subprocess
+import pytest
+from pytest import raises
+
+from Bio import Phylo
+from Bio.Phylo import BaseTree
+from Bio.Phylo.Consensus import _BitString
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference__init_.py:12]: This should be together in their own block.

File: Tests/my_test_Inference__init_.py [hunk 11]
```diff
@@ -0,0 +1,206 @@
+from contextlib import nullcontext as does_not_raise
```
Comment (JAlvarezJarreta) [inline Tests/my_test_Inference__init_.py:1]: Missing license header.

**Review by JAlvarezJarreta — CHANGES_REQUESTED** [review]
Besides the suggestions, it may also be useful to run `black` over these files.

**Comment by SMoragon** [comment]
I thank you the suggestions, but this was just for testing the tests. 
The real ones are in it's matching branch, "finished_tests".

**Pull request closed without merging**